docs/content/concepts/assets/asset-auto-execution.mdx

```diff
@@ -149,52 +194,61 @@ def asset1():
     ...
 ```
 
-## Auto-materialization and partitions
-
-Partitioned assets can have <PyObject object="AutoMaterializePolicy" pluralize />. Partitions are eligible for auto-materialization when either:
+### Auto-materialize policies and freshness policies
 
-- They are missing - i.e. have never been materialized before
-- New data has arrived in the partitions that are upstream of them
+<Note>
+  {" "}
+  Lazy auto-materialize policies do not currently have any effect on partitioned
+  assets.{" "}
+</Note>
 
-For time-partitioned assets (daily, hourly, etc.), only the last partition will be auto-materialized. To materialize earlier partition, launch a backfill.
+Instead of auto-materializing downstream assets immediately after new upstream data arrives, you can use <PyObject object="AutoMaterializePolicy" method="lazy" /> to materialize assets only when needed to meet an asset's <PyObject object="FreshnessPolicy" />. This allows you to avoid unnecessary materializations.
 
-Here's a pipeline with two daily-partitioned assets that have eager auto-materialize policies. At the end of each day, a partition for that day will be added to the set of partitions for each of the assets. Dagster will notice that the new partitions exist, but have no materializations, and then auto-materialize them.
+In this example, even if `asset1` is materialized every hour, `asset2` will only be materialized roughly once per day:
 
-```python file=/concepts/assets/auto_materialize_time_partitions.py
-from dagster import AutoMaterializePolicy, DailyPartitionsDefinition, asset
+```python file=/concepts/assets/auto_materialize_lazy.py
+from dagster import AutoMaterializePolicy, FreshnessPolicy, asset
 
 
-@asset(
-    partitions_def=DailyPartitionsDefinition(start_date="2020-10-10"),
-    auto_materialize_policy=AutoMaterializePolicy.eager(),
-)
+@asset
 def asset1():
     ...
 
 
 @asset(
-    partitions_def=DailyPartitionsDefinition(start_date="2020-10-10"),
-    auto_materialize_policy=AutoMaterializePolicy.eager(),
+    auto_materialize_policy=AutoMaterializePolicy.lazy(),
+    freshness_policy=FreshnessPolicy(maximum_lag_minutes=24 * 60),
     deps=[asset1],
 )
 def asset2():
     ...
 ```
 
-If the last partition of `asset1` is re-materialized, e.g. manually from the UI, then the corresponding partition of `asset2` will be auto-materialized after.
+Freshness policies express how fresh an asset needs to be relative to data at the root of the graph. This means that, for an asset to meet it's freshness policy, both it and upstream assets need to be materialized in time.
```

Comment:
Shouldn't this explanation of freshness policies come at the beginning of the section? I think since we've already explained auto-materialize policies in detail, context about freshness policies is what's needed first in this section.